web/packages/teleport/src/DesktopSession/useTdpClientCanvas.tsx

```diff
@@ -156,6 +166,36 @@ export default function useTdpClientCanvas(props: Props) {
     setWsConnection('open');
   };
 
+  const getModifierState = (e: KeyboardEvent, keyArg: string): ButtonState => {
+    let state = ButtonState.UP;
+    if (e.getModifierState(keyArg)) {
+      state = ButtonState.DOWN;
+    }
+    return state;
+  };
+
+  const getSyncKeys = (e: KeyboardEvent): SyncKeys => {
+    return {
+      scrollLockState: getModifierState(e, 'ScrollLock'),
+      numLockState: getModifierState(e, 'NumLock'),
+      capsLockState: getModifierState(e, 'CapsLock'),
+      kanaLockState: getModifierState(e, 'KanaMode'),
```

Comment:
It's definitely `KanaMode` and not `KanaLock`? Is there a reference we can link to?

Reply:
Yes although it looks like it might be "Lang1" on mac: https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_code_values

Reply:
Good question, as it turns out I was looking at incorrect documentation. The [official w3 documentation](https://www.w3.org/TR/uievents-key/#keys-modifier) doesn't advertise support for `KanaMode` (or `KanaLock`, or anything Kana-related under Modifier Keys).

I think the best course of action is to just hardcode it to `UP`, that will give an optimal experience to the majority of users and we can revisit if desktop access really takes off in Japan.

Reply:
4bf92fc02f893cb5d8c976abe3e65e2d2e521cd0